articles/whats-wrong-with-i-dont-write-any-tests-i-am-not-a-tester.md

```diff
@@ -0,0 +1,90 @@
+Henri (made up name): "I don’t write any tests since I am not a tester".
+
+I know he handled the situation quite well, but he asked me for advice.
+
+One thing to consider is the underlying question to this, which might be:
+"How do we get people to change their behavior?"
+So here are my thoughts.
+
+## **Yourself First**   
+First of all, think about things to consider for yourself:
+* Why is the practice or tool that you are suggesting any better than the current way of doing things?
+* Can you explain the value in the proposed change?
+* Can you lead by example?
+* Do you have enough patience and skills to teach others?
+I would try to work on yourself before trying to change others.
+    
+ 
+## **Roles?**  
+I see that a lot of people are focused on their role, forgetting the bigger picture of the team and purpose of the work they are doing.
+I would ask these questions:
+* Are we one team that focuses on the Sprint Goal?
+  In a Scrum context there is no “tester”, “programmer” or “architect”, we are all professional engineers that deliver value through collaboration.
+* No matter what, do we stand together and support each other?
+* Are we doing whatever is necessary to deliver a usable product every Sprint?
+ 
+
+## **Done?**  
+Henri: "I don’t write any tests since I am not a tester".  
+Ask: "How do you know when you are done?"  
+  
+What is on our Definition of Done?
+How can we build a usable, tested and fully integrated product increment every Sprint?
+Are we doing that already? Why not?
+   
+
+## **Tests are as important as documentation**  
+Documentation is needed, and one good way to document how software **must** work are tests. I emphasize **must**, since documentation only documents how the software **might** work.  
+We learned too often that documentation gets out of sync too easily.  
+I blogged about the ["Why are *automated tests* so important?"](http://blog.gfader.com/2010/10/why-are-automated-tests-so-important.html) on my personal blog.  
+
+   
+
+## **Quality**  
+Quality is everyone’s responsibility in a Scrum Team. There is no QA Team in a Scrum context, which means the whole Scrum team is responsible for delivering high quality software that works and is fully tested.  
+  
+Quality attributes that are important:
+* Does it work at all?  
+  Huh!
+* Does it work well?
+* Is it deployed and use-able?  
+  Are the users able to access it?
+* Is it useful? 
+* Is it successful?
+* Does it make the impact we wanted to achieve?  
+   -> Yes! Value is key
+
+   
+## **Tests are code**  
+Are you a coder?? Yes?   
+Tests are code. So write some tests, especially so that you can sleep better at night.  
+  
+
+## **Still not willing to write tests?**  
+Ask: 
+* How can You help? 
+* What can You do?  
+  
+You can always get them coffee.
+  
+Show support. We are in this together.  
+  
+*By {{page.authorName}}*
+
```

Comment:
I think most important argument is somehow missing. I think automated tests are primarily something to help developers find their mistakes while they code by giving them fast automated feedback. That is one of the most important reasons to automate tests. And therefore it is kind of obvious that developers also need to care about those tests and be able to adjust them when they adjust some functionality. Otherwise these tests become outdated and useless very quickly and will not give the developers any valuable feedback anymore while they are coding. Investing in such a kind of test automation that is not useful for the developers while implementing because they did not write it or can not adjust it seems very useless to me. 

And: I often used to hear something else but very similar: "I do not need to write tests, I write code that works.".
If I hear that I usually answer "Yes, but since you work in a team and your colleagues are not as clever as you are (or as you think you are), you should write at least tests to help your colleagues not to break your beautiful and perfect piece of code" ;-)

See also some slides by me about same topic here:
http://scenarioo.org/e2e-testing-with-scenarioo-js/#/motivation-my-code-works
(use arrow down to move through next around 7 slides of that chapter)

Maybe you can somehow put those points into the article??

Apart form that, I think the article is cool and its a very important topic!

Reply:
Thanks Rolf. Good catch. I focused mainly on the „Tester“ role or activity and didn’t think that much about: „How do you know you are done?“ or „How do you know it works in 5 years“. 
(Tests are the answer).

I will add that.
THANKS!!!